scripts/globals/job_utils/dancer.lua

```diff
@@ -62,6 +76,14 @@ local function getStepFinishingMovesBase(player)
         numAwardedMoves = 2
     end
 
+    -- Terpsichore main-hand FM bonus.
+    -- TODO: Confirm in retail the bonus only applies when the weapon is equiped in main hand.
+    local mainHandWeapon = player:getEquipID(xi.slot.MAIN)
+
+    if terpsichoreTable[mainHandWeapon] then
```

Comment:
Steps != Finishing moves.  Steps are the number of applications of a specific type of step (Cap is 10, regardless)